### PR TITLE
Feat/kui 961 add application code

### PR DESCRIPTION
**Pull request by 4886Talga**
please review the code

### CODE REVIEW

File: common-context/createCommonContextFunctions.js [hunk 1]
```diff
@@ -83,17 +83,19 @@ function handleCourseData(courseObject, courseCode, userName, language) {
...
-          startDate: round.firstTuitionDate,
-          ladokUID: round.ladokUID,
-          canBeAccessedByUser: resolveUserAccessRights(this.member, round, this.courseCode, semester),
-        })
-      })
+      thisStore.roundData[semester] = semesterRounds.map(
+        round =>
+          (round.applicationCode = {
+            courseCode: this.courseCode,
+            roundId: round.ladokRoundId,
```
Comment (lenaar) [inline common-context/createCommonContextFunctions.js:90]: `roundId: round.ladokRoundId` has to be removed

File: public/js/app/components/MemoMenu.jsx [hunk 2]
```diff
@@ -30,9 +30,9 @@ function MemoMenu(props) {
       cancel: false,
     },
     semester: initSemester && initSemester.length > 0 ? initSemester : '',
-    rounds: initTempData ? initTempData.roundIdList : [],
-    usedRounds: context.usedRounds.usedRoundsIdList || [],
-    usedRoundsWithWebVer: context.usedRounds.roundsIdWithWebVersion || {},
+    rounds: initTempData ? initTempData.applicationCodes : [],
+    usedRounds: context.usedRounds.usedRoundsApplicationCodeList || [],
```
Comment (lenaar) [inline public/js/app/components/MemoMenu.jsx:34]: Is `kurs-pm-data-api` reflect these name changes in `usedRounds`? Will it have `usedRoundsApplicationCodeList` and `roundsApplicationCodeWithWebVersion`?

File: server/apiCalls/koppsCourseData.js [hunk 3]
```diff
@@ -14,9 +14,42 @@ const koppsApi = new BasicAPI({
   defaultTimeout: 10000, // config.koppsApi.defaultTimeout
 })
 
+async function _getApplicationCodeFromLadokUID(ladokUID) {
+  try {
+    log.debug('Going to fetch application for ladokUID: ', ladokUID)
+    const res = await koppsApi.getAsync(`courses/offerings/roundnumber?ladokuid=${ladokUID}`)
```
Comment (lenaar) [inline server/apiCalls/koppsCourseData.js:20]: Destructure body direct here,  `const { body } = await koppsApi.getAsync(...)`

File: server/apiCalls/koppsCourseData.js [hunk 4]
```diff
@@ -14,9 +14,42 @@ const koppsApi = new BasicAPI({
...
+    }
+  } catch (error) {
+    log.error('Kopps is not available', error)
+  }
+}
+
 async function getKoppsCourseData(courseCode) {
   try {
-    return await koppsApi.getAsync(`course/${encodeURIComponent(courseCode)}/courseroundterms`)
+    const res = await koppsApi.getAsync(`course/${encodeURIComponent(courseCode)}/courseroundterms`)
```
Comment (lenaar) [inline server/apiCalls/koppsCourseData.js:35]: Destructure body direct here,  `const { body } = await koppsApi.getAsync(...)`

File: test/mocks/mockWebContext.js [hunk 5]
```diff
@@ -36,6 +36,7 @@ const mockWebContext = (userLang = 'en') => {
           ladokUID: 'd',
           language: 'English',
           roundId: '2',
```
Comment (lenaar) [inline test/mocks/mockWebContext.js:38]: `roundId` has to be removed in all tests

**Review by lenaar — CHANGES_REQUESTED** [review]
Code need some cleaning of `roundId` and double check if changes here reflects changes in api

File: common-context/createCommonContextFunctions.js (hunk 1 above)
Comment (Rao-Sabahat-Ali) [reply common-context/createCommonContextFunctions.js]: Done

File: server/apiCalls/koppsCourseData.js (hunk 3 above)
Comment (Rao-Sabahat-Ali) [reply server/apiCalls/koppsCourseData.js]: Done

File: server/apiCalls/koppsCourseData.js (hunk 4 above)
Comment (Rao-Sabahat-Ali) [reply server/apiCalls/koppsCourseData.js:34]: Done

File: test/mocks/mockWebContext.js (hunk 5 above)
Comment (Rao-Sabahat-Ali) [reply test/mocks/mockWebContext.js]: Done

File: public/js/app/components/MemoMenu.jsx (hunk 2 above)
Comment (Rao-Sabahat-Ali) [reply public/js/app/components/MemoMenu.jsx:34]: This data is coming from `kurs-pm-api`. There it is updated

**Review by lenaar — APPROVED** [review]
Looks good to me

**Review by lenaar — APPROVED** [review]
Looks good to me

**Pull request merged**